Make rustc shim's verbose output include crate_name being compiled.

pnkfelix opened the pull request:
This change is mainly motivated by an issue with the environment printing I added in PR 82403: multiple rustc invocations progress in parallel, and the environment output, spanning multiple lines, gets interleaved in ways make it difficult to extra the enviroment settings.

(This aforementioned difficulty is more of a hiccup than an outright show-stopper, because the environment variables tend to be the same for all of the rustc invocations, so it doesn't matter too much if one mixes up which lines one is looking at. But still: Better to fix it.)

Mark-Simulacrum commented:
Do you care about cargo test? We emit test:false/true for the rustc-timing lines

pnkfelix commented:
> Do you care about cargo test? We emit test:false/true for the rustc-timing lines

I don't understand your question.

Are you asking whether I should consider adding such contextual info (the boolean value of `test`) to the output I'm adding here?

Or are you saying I may want to add similar contextual info elsewhere? (Note that the `[RUSTC-TIMING]` lines already include the crate-name, which must be `Some` for those lines to be emitted in the first place...)

(I didn't do a thorough review of where else in this code such contextual info might be of use. I wanted to address the more immediate problem of this particular series of lines and how confusing they can be when many instances are running in parallel.)

Mark-Simulacrum commented:
Sorry, I'm saying - if you care about distinguishing between rustc and rustc --test invocations, then you should add that information to the output.

pnkfelix commented:
Took @Mark-Simulacrum 's advice and added `--test` to output when it was part of the given arguments.

Also added `[RUSTC-SHIM]` to the prefix so that its more apparent where these lines originate. (Its easy enough for me to M-x kill-rectangle them, and I assume others can regexp delete them in a similar manner when desired.)

Here is what some sample bits of output look like now:

```
...
[RUSTC-SHIM] rustc rustc_driver env[40]: "RUSTFLAGS"="-Zmacro-backtrace -Clink-args=-Wl,-rpath,$ORIGIN/../lib -Ztls-model=initial-exec -Zunstable-options -Wrustc::internal -Cprefer-dynamic"
[RUSTC-SHIM] rustc rustc_driver env[41]: "RUST_TEST_THREADS"="128"
[RUSTC-SHIM] rustc rustc_driver working directory: /media/pnkfelix/Rust/rust.git
...
[RUSTC-SHIM] rustc --test rustc_main env[0]: "CARGO"="/media/pnkfelix/Rust/rust.git/objdir-default/build/x86_64-unknown-linux-gnu/stage0/bin/cargo"
[RUSTC-SHIM] rustc --test rustc_main env[1]: "CARGO_BIN_NAME"="rustc-main"
[RUSTC-SHIM] rustc --test rustc_main env[2]: "CARGO_CRATE_NAME"="rustc_main"
...
```

Mark-Simulacrum commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 444a7567693506347a3ce9e79f6823abf7141b17 has been approved by `Mark-Simulacrum`

<!-- @bors r=Mark-Simulacrum 444a7567693506347a3ce9e79f6823abf7141b17 -->
<!-- homu: {"type":"Approved","sha":"444a7567693506347a3ce9e79f6823abf7141b17","approver":"Mark-Simulacrum"} -->
